src/script/script.h

```diff
@@ -571,16 +571,34 @@ class CScript : public CScriptBase
         if (b.empty())
             return nFound;
         iterator pc = begin();
+        iterator copied = begin();
+        size_t nShift = 0;
         opcodetype opcode;
         do
         {
             while (end() - pc >= (long)b.size() && memcmp(&pc[0], &b[0], b.size()) == 0)
             {
-                pc = erase(pc, pc + b.size());
+                // Lazy copy-in-place if there is a match:
+                while (copied != pc-nShift) {
+                    *copied = *(copied+nShift);
+                    ++copied;
+                }
+                pc += b.size();
+                nShift += b.size();
                 ++nFound;
             }
         }
         while (GetOp(pc, opcode));
+
+        if (nShift > 0)
+        {
+            while (copied != pc-nShift) {
```

Comment:
I think this changes the result of FindAndDelete for invalid CScripts that end in a pushdata without enough data. Replacing "pc" with "end()" would keep the original behavior in this case.